### PR TITLE
feat(eventBroker): eventBroker acts in the bubbling phase

### PR DESCRIPTION
**Pull request by Mishnya**
Collecting events at the capture phase instead of the bubbling phase allows to get information before any effects occur on the event.

### CODE REVIEW

File: src/components/Link/Link.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ export interface LinkProps extends DOMProps, QAProps {
 
 const b = block('link');
 
-const PureLink = React.forwardRef<HTMLAnchorElement | HTMLSpanElement, LinkProps>(function Link(
+export const Link = React.forwardRef<HTMLElement, LinkProps>(function Link(
```
Comment (amje) [inline src/components/Link/Link.tsx:28]: This can break types

File: src/components/Menu/MenuItem.tsx [hunk 2]
```diff
@@ -21,7 +21,7 @@ export interface MenuItemProps extends DOMProps, QAProps {
     children?: React.ReactNode;
 }
 
-const PureMenuItem = React.forwardRef<HTMLLIElement, MenuItemProps>(function MenuItem(
+export const MenuItem = React.forwardRef<HTMLElement, MenuItemProps>(function MenuItem(
```
Comment (amje) [inline src/components/Menu/MenuItem.tsx:24]: types

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/Link/Link.tsx (hunk 1 above)
Comment (Mishnya) [reply src/components/Link/Link.tsx:28]: `withEventBrokerDomHandlers.tsx` uses `React.forwardRef<HTMLElement, ...>` to wrap components.
So not changing types to `React.forwardRef<HTMLElement, ...>` breaks types.

File: src/components/Menu/MenuItem.tsx (hunk 2 above)
Comment (Mishnya) [reply src/components/Menu/MenuItem.tsx:24]: Same with `withEventBrokerDomHandlers`

File: src/components/Link/Link.tsx (hunk 1 above)
Comment (amje) [reply src/components/Link/Link.tsx:28]: Oh, ok

**Review by amje — APPROVED** [review]

**Review by korvin89 — APPROVED** [review]

**Review by LakeVostok — APPROVED** [review]

**Pull request merged**